plugins/destination/plugin.go

```diff
@@ -65,6 +49,12 @@ type Client interface {
 	Close(ctx context.Context) error
 }
 
+type BatchingWriter interface {
```

Comment:
This can just be an `UnmanagedWriter` but kept as this for development/clarity purposes.
`type UnmanagedWriterFunc func(ManagedWriter) UnmanagedWriter` line would look a bit weird.